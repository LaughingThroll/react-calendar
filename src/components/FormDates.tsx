import React from "react"
import { TChildren } from "../types/utilsTypes"

interface IFormDates {
  title?: string
  inner?: boolean
  children?: TChildren
}

const FormDates: React.FC<IFormDates> = ({ title, inner, children }) => {
  return (
    <>
      <div className="form-dates">
        <h5 className="form-dates__title">{title}</h5>
        {inner ? <div className="form-dates__inner"> {children}</div> : children}
      </div>
    </>
  )
}

export default FormDates
