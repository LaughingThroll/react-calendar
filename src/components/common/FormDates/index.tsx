import React from 'react'
import styles from './formDates.module.scss'
import { TChildren } from '../../../types/utilsTypes'

export interface FormDatesProps {
  title?: string
  inner?: boolean
  children?: TChildren
}

const FormDates: React.FC<FormDatesProps> = ({ title = 'Title', inner, children }) => {
  const { formDates, formDates__title, formDates__inner } = styles

  return (
    <div className={formDates}>
      <h5 className={formDates__title}>{title}</h5>
      {inner ? <div className={formDates__inner}> {children}</div> : children}
    </div>
  )
}

export default React.memo(FormDates)
