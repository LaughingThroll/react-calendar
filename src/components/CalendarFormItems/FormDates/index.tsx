import React from 'react'
import { FormItem, InputDate } from '../../common'

import { ControlForInput } from '../../../types/controls'

export interface FormDatesProps {
  startDate: ControlForInput
  endDate: ControlForInput
  classNames?: string[]
}

const FormDates: React.FC<FormDatesProps> = ({ startDate, endDate, classNames = [] }) => {
  return (
    <FormItem title="Dates" classNames={classNames} inner>
      <InputDate label="From" onChange={startDate.changeValue} value={startDate.value} name="startDate" />
      <InputDate label="To" onChange={endDate.changeValue} value={endDate.value} name="endDate" />
    </FormItem>
  )
}

export default React.memo(FormDates)
