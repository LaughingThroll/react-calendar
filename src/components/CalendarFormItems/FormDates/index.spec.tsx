import { shallow } from 'enzyme'
import FormDates, { FormDatesProps } from './'

const setUp = (props: FormDatesProps) => shallow(<FormDates {...props} />)

describe('FormDates render', () => {
  it('default render', () => {
    const props: FormDatesProps = {
      startDate: { changeValue: jest.fn(), value: '' },
      endDate: { changeValue: jest.fn(), value: '' },
    }
    const tree = setUp(props)

    expect(tree).toMatchSnapshot()
  })
})
