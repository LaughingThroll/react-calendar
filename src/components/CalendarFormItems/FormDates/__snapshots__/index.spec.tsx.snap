// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FormDates render default render 1`] = `
<Memo(FormItem)
  classNames={Array []}
  inner={true}
  title="Dates"
>
  <Memo(InputDate)
    label="From"
    name="startDate"
    onChange={[MockFunction]}
    value=""
  />
  <Memo(InputDate)
    label="To"
    name="endDate"
    onChange={[MockFunction]}
    value=""
  />
</Memo(FormItem)>
`;
